rskj-core/src/main/java/co/rsk/scoring/PeerScoringReporterService.java

```diff
@@ -0,0 +1,79 @@
+package co.rsk.scoring;
+
+import co.rsk.config.InternalService;
+import com.google.common.annotations.VisibleForTesting;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * This service prints a summary of nodes with bad reputation, every 5 minutes (configurable)
+ * */
+public class PeerScoringReporterService implements InternalService {
+
+    private static final Logger logger = LoggerFactory.getLogger("peerScoring");
+
+    private final PeerScoringManager peerScoringManager;
+    private final long time;
+    private ScheduledExecutorService scheduledExecutorService = Executors.newSingleThreadScheduledExecutor();
+    private boolean running = false;
+
+    public PeerScoringReporterService(long time, PeerScoringManager peerScoringManager) {
+        this.time = time;
+        this.peerScoringManager = peerScoringManager;
+    }
+
+    @Override
+    public void start() {
+        logger.debug("starting peer scoring reporter service");
+        try {
+            List<PeerScoringInformation> peerScoringInformationList = peerScoringManager.getPeersInformation();
+            scheduledExecutorService.scheduleAtFixedRate(() -> printReport(peerScoringInformationList),
+                    0,
+                    time,
+                    TimeUnit.MILLISECONDS
+            );
+            running = true;
+        } catch (Exception e) {
+            logger.warn("peer scoring reporter failed", e);
+            stop();
+        }
+    }
+
+    @Override
+    public void stop() {
+        scheduledExecutorService.shutdown();
+        logger.warn("peer scoring reporter service has been stopped");
+        running = false;
+    }
+
+    public boolean printReport(List<PeerScoringInformation> peerScoringInformationList) {
```

Comment:
As for me this error-handling is a bit over-complicated. Result of this method is not being used by the `scheduledExecutorService`, is it? It's only for testing purposes? Can we alternatively check if this method throws an exception in the tests, and return nothing here? The `PeerScoringReporterUtil`'s utility methods can only produce unchecked/runtime exceptions I suppose.

Reply:
yep you're right, I confused Exceptions with `RuntimeExceptions`. I'll move testing logic to a `TestPeerScoringReporterService` as @ajlopez suggested yesterday.

Reply:
https://github.com/rsksmart/rskj/pull/1351/commits/9b301af388db91a3dae7d6c7860f1b5dc7a4291a